tests/platform/broadcom/files/ser_injector.py

```diff
@@ -0,0 +1,306 @@
+                            u'L3_ENTRY_IPV6_UNICAST.ipipe0', u'L3_DEFIP_ALPM_IPV4.ipipe0', u'FP_GLOBAL_MASK_TCAM.ipipe0',
+                            u'FP_STORM_CONTROL_METERS.ipipe0', u'FP_GM_FIELDS.ipipe0']
+
+# Not every memory table orrection is detected by the test case. This could be due to the
+# losses over syslog transport. Make the test iterate over different memory tables in order to cover
+# every memory table
+DEFAULT_TEST_ITERATION = 3
+
+def run_cmd(cmd):
+    '''
+    @summary: Utility that runs a command in a subprocess
+    @param cmd: Command to be run
+    @return: stdout of the command run
+    @return: stderr of the command run
+    '''
+    out = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    stdout, stderr = out.communicate()
+    return stdout, stderr
+
+class BcmMemory():
+    '''
+    @summary: BcmMemory captures different memory tables of the Broadcom ASIC. Memory are split into two categories:
+              cached and uncached. Broadcom SER correction is enabled for cached memory tables. For cached memory tables,
+              memory attributes are also retreived
+    '''
+    def __init__(self):
+        '''
+        @summary: Class constructor
+        '''
+        self.cached_memory = {}
+        self.uncached_memory = {}
+        self.memory_address = {}
+
+    def get_memory_attributes(self, mem):
+        '''
+        @summary: Reads Broadcom memory attributes using list command. Attributes include start address, flags, 
+                  number of entries, entry size in bytes and entry size in words. The method uses regex to parse
+                  the command output since there is not SAI APIs for it.
+        '''
+        stdout, stderr = run_cmd(["bcmcmd",  "list " + mem])
+
+        attributes = stdout.decode("utf-8").split("\n")
+
+        attr = {}
+        m = re.search('^Memory:.*address (.+)$', attributes[1])
+        attr['address'] = int(m.group(1), 16)
+
+        m = re.search('^Flags: (.*)$', attributes[2])
+        attr['flags'] = m.group(1).strip().split(" ")
+
+        m = re.search('^Entries: (\d+).*each (\d+) bytes (\d+) words', attributes[4])
+        attr['entries'] = int(m.group(1))
+        attr['size_byte'] = int(m.group(2))
+        attr['size_word'] = int(m.group(3))
+
+        return attr
+
+    def read_memory(self):
+        '''
+        @summary: Read different memory tables using cache command. It update both cached_memory and uncached_memory
+                  hash tables. For cached memory, ut aksi creat a reverse index of address to memory table name. This indez
+                  is stored in memory_address hash table
+        '''
+        stdout, stderr = run_cmd(['bcmcmd', 'cache'])
+
+        cache_flag = False
+        memories = stdout.decode("utf-8").split("\n")
+
+        # remove Head line and 3 trailing prompt lines 
```

Comment:
can you please add example of entries you're parsing here?

Reply:
Yes, Added sample output